### PR TITLE
chore(deploy) bump Kong to 3.1

### PR DESCRIPTION
**Pull request by rainest**
**What this PR does / why we need it**:

It's [out](https://hub.docker.com/layers/kong/kong-gateway/3.1.0.0-debian/images/sha256-ad345462fe02a5acec549ad0e6d6f56d1e479ca41f5fcdb5366f971bb8bdfa9f?context=explore). Part of  #3220

**PR Readiness Checklist**:

Complete these before marking the PR as `ready to review`:

- [ ] ~the `CHANGELOG.md` release notes have been updated to reflect any significant (and particularly user-facing) changes introduced by this PR~

### CODE REVIEW

**Comment by rainest** [comment]
E2E run: https://github.com/Kong/kubernetes-ingress-controller/actions/runs/3735000207

File: config/image/enterprise/kustomization.yaml [hunk 1]
```diff
@@ -6,7 +6,7 @@ kind: Component
 images:
 - name: kong
   newName: kong/kong-gateway
-  newTag: '3.0'
+  newTag: '3.1'
 - name: kong-placeholder
   newName: kong/kong-gateway
   newTag: '3.0'
```
Comment (randmonkey) [inline config/image/enterprise/kustomization.yaml:12]: should here also be updated to `3.1`?

Comment (pmalek) [inline config/image/enterprise/kustomization.yaml:12]:
```suggestion
  newTag: '3.1'
```

Comment (pmalek) [reply config/image/enterprise/kustomization.yaml:12]: I believe so, yes. Adding a change with this to this PR

**Review by pmalek — APPROVED** [review]

**Pull request merged**